src/shared/osclib/__init__.py

```diff
@@ -152,115 +152,44 @@ def are_on_same_machine(url1: str, url2: str) -> bool:
-                if socket.gethostbyaddr(addr1) == socket.gethostbyaddr(addr2):
-                    return True
-
-            except BaseException as e:
-                _logger.warning(
-                    f'Failed to find host by addr'
-                    f'for "{addr1}" or "{addr2}"\n'
-                    f'{str(e)}')
-
-        except BaseException as e:
-            _logger.warning(
-                f'Failed to find host by name for '
-                f'"{hostname_1}" or "{hostname_2}"\n'
-                f'{str(e)}')
-            
-    # if (socket.gethostbyname(address1.hostname)
-    #             in ('127.0.0.1', '127.0.1.1')
-    #         and socket.gethostbyname(address2.hostname)
-    #             in ('127.0.0.1', '127.0.1.1')):
-    #     return True            
-    
-    elif addr1_ipv6:
-        try:
-            if (socket.gethostbyaddr(hostname_1)[0] == 'localhost'
-                    and socket.gethostbyname(hostname_2)
-                        in ('127.0.0.1', '127.0.1.1')):
-                return True
-
-        except BaseException as e:
-            _logger.warning(str(e))
-            
-    elif addr2_ipv6:
-        try:
-            if (socket.gethostbyaddr(hostname_2)[0] == 'localhost'
-                    and socket.gethostbyname(hostname_1)
-                        in ('127.0.0.1', '127.0.1.1')):
-                return True
+    LOCAL_ADDRS = ('127.0.0.1', '127.0.1.1', '::1')
 
-        except BaseException as e:
-            _logger.warning(str(e))
+    if host1 in LOCAL_ADDRS and host2 in LOCAL_ADDRS:
+        return True
 
     ip = get_machine_192()
 
-    print('are_on_same_machine', url1, url2)
-    print(f"    '{ip}' '{address1.hostname}' '{address2.hostname}")
-
-    if ip not in (hostname_1, hostname_2):
+    if ip not in (resolve_host(address1.hostname), resolve_host(address2.hostname)):
         return False
 
-    try:
-        if (ip == socket.gethostbyname(hostname_1)
-                == socket.gethostbyname(hostname_2)):
-            # on some systems (as fedora),
-            # socket.gethostbyname returns a 192.168.. url
-            return True
-
-    except BaseException as e:
-        _logger.warning(str(e))
-
-    try:
-        if (socket.gethostbyname(hostname_1)
-                in ('127.0.0.1', '127.0.1.1')):
-            if hostname_2 == ip:
-                return True
-
-    except BaseException as e:
-        _logger.warning(str(e))
+    if ip == resolve_host(address1.hostname) == resolve_host(address2.hostname):
+        # on some systems (as fedora),
+        # socket.gethostbyname returns a 192.168.. url
+        return True
 
-    try:
-        if (socket.gethostbyname(hostname_2)
-                in ('127.0.0.1', '127.0.1.1')):
-            if hostname_1 == ip:
-                return True
-    except BaseException as e:
-        _logger.warning(str(e))
+    for host in (address1.hostname, address2.hostname):
+        if resolve_host(host) in LOCAL_ADDRS and host == ip:
```

Comment:
already in original code, but I think to this... There is probably no reason that 'resolve_host' returns a different value at each call (with the same argument. So a dict could be stocked directly somewhere in __init__.py. and resolve_host could return directly the matching value of the dict if it exists (faster execution).

Reply:
technically, if two hosts communicate each other, and one of them is loopback interface, then another interface is on the same machine and we can make decision this is the same machine. But I can not explain the double checking in original code, so I decided to save this double checking logic untouched.

Reply:
I can't explain it too ;) Probably because it is a sensible function, because I can't really see its impact in many cases. I may have  chosen to modify the strict minimum, and I have not seen the double checking.